rust_language/generate.go

```diff
@@ -400,6 +413,45 @@ func (l *rustLang) generateCargoRule(c *config.Config, args *language.GenerateAr
 		newRule.SetAttr("crate_root", *crateRoot)
 	}
 
+	dependsOnBuildScript := false
+	if kind == "rust_library" || kind == "rust_binary" {
+		for _, src := range args.RegularFiles {
+			if src == "build.rs" {
+				dependsOnBuildScript = true
+			}
+		}
+	}
```

Comment:
I think it would be cleaner to move this check into the parent function (`generateRulesFromCargo`) so it only needs to happen once.